### PR TITLE
interface: better checking if channel should be marked unread

### PR DESCRIPTION
**Pull request by tylershuster**
Basically, check if the line is visible after the initialization period has passed, after which point we should know if the blue line is visible. Worked in testing for me.

Also some style nits and removed `console.log`s

### CODE REVIEW

File: pkg/interface/src/views/apps/chat/components/ChatWindow.tsx [hunk 1]
```diff
@@ -136,6 +135,10 @@ export default class ChatWindow extends Component<ChatWindowProps, ChatWindowSta
       this.setState({ fetchPending: false });
     }
 
+    if (this.state.initialized !== prevState.initialized) {
+      this.dismissIfLineVisible();
+    }
```
Comment (tylershuster) [inline pkg/interface/src/views/apps/chat/components/ChatWindow.tsx:140]: This is the relevant bit

Comment (matildepark) [reply pkg/interface/src/views/apps/chat/components/ChatWindow.tsx:140]: Won't this fire if state.initialized is `false` and prevState's is `true`?

Comment (tylershuster) [reply pkg/interface/src/views/apps/chat/components/ChatWindow.tsx]: I don't think that would ever happen, but if there's some weirdness with switching chat windows, I changed it to make sure

**Comment by matildepark** [comment]
Haven't stamped this as of yet because I can't quite replicate the problem. On my ship I see the new messages banner until I scroll up while the line is on screen, then it vanishes. Is this case best tested with "the chat has less than 20 messages" case?

**Comment by tylershuster** [comment]
Does that behavior still occur with this patch applied?

**Comment by matildepark** [comment]
Actually, yeah, it never gets marked as read and there's no line visible.

**Comment by tylershuster** [comment]
Needs more systematic attention, and I deleted this fork.

**Pull request closed without merging**